dev/tasks/r/github.nightly.yml

```diff
@@ -0,0 +1,381 @@
+      version: {{ '${{ steps.save-version.outputs.version }}' }}
+      date: {{ '${{ steps.save-version.outputs.date }}' }}
+    steps:
+      {{ macros.github_checkout_arrow()|indent }}
+      {{ macros.change_r_pkg_version()|indent }}
+      - name: Save Version
+        id: save-version
+        shell: bash
+        run: | 
+          echo "::set-output name=version::$(grep ^Version arrow/r/DESCRIPTION | sed s/Version:\ //)"
+          echo "::set-output name=date::$(date +%Y%m%d)"
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          install-r: false
+
+      - name: Build R source package
+        shell: bash
+        run: |
+          cd arrow/r
+          # Copy in the Arrow C++ source
+          make sync-cpp
+          R CMD build --no-build-vignettes .
+
+      - name: Install davfs2 & Mount Repo
+        run: | 
+          sudo apt update && sudo apt install davfs2
+          mkdir nightly
+          sudo bash -c 'echo "{{ webdav_domain }} {{ '${{ secrets.CROSSBOW_NIGHTLIES_USER }} ${{ secrets.CROSSBOW_NIGHTLIES_TOKEN }}' }}" >> /etc/davfs2/secrets'
+          sudo mount -t davfs {{ webdav_domain }} nightly -o rw
+
+      - name: Upload Source Package
+        run: | 
+          # ensure repo structure is set up, this job is the fastest and should fix any issue before the other jobs try
+          # to push into non existent folder with curl, which would fail silently. 
+          sudo mkdir -p {{ r_root }}/src/contrib 
+        {% for os in ["ubuntu-18.04", 
+                      "centos-7",
+                      "windows"] %}     
+          sudo mkdir -p {{ r_root }}/libarrow/bin/{{ os }}
+        {% endfor %}
+        {% for os in ["windows", "macosx", "macosx/big-sur-arm64"] %}     
+          {% for r_version in ["4.1", "4.2"] %}
+          sudo mkdir -p {{ r_root }}/bin/{{ os }}/contrib/{{ r_version }}
+          {% endfor %}
+        {% endfor %}
+        
+          sudo cp arrow/r/arrow_*.tar.gz {{ r_root }}/src/contrib
+      - name: Update Repo
+        shell: sudo Rscript {0}
+        run: |
+          if(file.exists("{{ r_root }}/src/contrib/PACKAGES")) {
+            tools::update_PACKAGES("{{ r_root }}/src/contrib" , type = "source", latestOnly = FALSE)
+          } else {
+            tools::write_PACKAGES("{{ r_root }}/src/contrib" , type = "source", latestOnly = FALSE)
+          }
+
+      # ensures all changes are written
+      - run: sudo umount nightly 
+      - name: Upload binary artifact (temp)
+        uses: actions/upload-artifact@v3
+        with:
+          name: r-src-pkg
+          path: arrow/r/arrow_*.tar.gz
+
+  linux-cpp:
+    name: C++ Binary {{ '${{ matrix.config.os }}-${{ matrix.config.version }}' }}
+    runs-on: ubuntu-latest
+    needs: source
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - { os: ubuntu, version: "18.04" }
+          - { os: centos, version: "7" }
+    env:
+      UBUNTU: {{ '${{ matrix.config.version }}' }}
+      R: 3.6

```

Comment:
Could we? Should we? bump this version?

Reply:
I added this from the arrow-r-nightly `.env` file and I am unsure if this is even needed as this job just compiles libarrow via `r/inst/build_arrow_static.sh` afaik no R needed for that right?

Reply:
Yeah, it would be worth trying to remove it entirely and see if it goes

Reply:
Yep, can be removed https://github.com/assignUser/test-repo-a/actions/runs/2362911923/workflow#L142